docs/core/Lifecycle.md

```diff
@@ -0,0 +1,102 @@
+# Carbon Lifecycle
+
+For a consistent development experience Carbon uses the same underlying interface for all graph types. All Carbon graphs are separated into 2 parts: `Construct` & `GraphContent`.
+
+-   [Carbon Lifecycle](#carbon-lifecycle)
+    -   [Construct](#construct)
+        -   [`loadContent(input)`](#loadcontentinput)
+        -   [`unloadContent(input)`](#unloadcontentinput)
+        -   [`destroy`](#destroy)
+    -   [Important information](#important-information)
+    -   [GraphContent](#graphcontent)
+
+## Construct
+
+`Construct` is the base for `Graph` and other types of graph that serves as a container for the datasets (content). Usually this takes care of one or more of the following:
+
+-   X Axis
+-   Y Axis
+-   Vertical Grid
+-   Horizontal Grid
+-   X Axis labels
+-   Y Axis labels
+-   Legend
+
+Construct exposes following functions to handle the graph or its contents:
+
+-   `generate` - Called internally when instantiating the graph
+-   `loadContent` - Load a dataset in the graph
+-   `unloadContent` - Remove a dataset from the graph
+-   `resize` - Resizes the graph based on window resize event
+-   `destroy` - Destroys the graph
+
```

Comment:
If the `Construct` phase has 5 lifecycle methods then can we also indicate them along with the rest on line# 7?